packages/core/test/support.ts

```diff
@@ -278,12 +278,17 @@ export function getPoolMax(): number {
   return Config[getTestDialect()].pool?.max ?? 1;
 }
 
-type ExpectationKey = 'default' | Permutations<Dialect>;
+type ExpectationKey = 'default' | Permutations<Dialect, 4>;
```

Comment:
Would it be possible to use something like `Math.floor(Dialect.length / 2)` instead of 4? Not sure if we can do these computations in type definitions though. Otherwise it might be useful to add a comment here to mention that 4 should be equal to half of the supported dialects.

Reply:
It isn't, and I also don't recommend automatically increasing it every time a dialect is added. Its complexity is exponential, I wouldn't recommend going past 4

If it becomes a problem the only alternative I can think of would be to disable this type altogether and rely on the runtime error

Reply:
We could see about introducing additional combinations with more dialects manually when the tests require them instead of computing all possible permutations. But I don't think that's necessary anytime soon so we can just stick to 4 for now then

Reply:
That's a good idea!